mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -232,7 +232,7 @@ public final int hashCode() {
      * @throws MailboxException
      *             when message cannot be appended
      */
-    AppendResult appendMessage(InputStream msgIn, Date internalDate, MailboxSession mailboxSession, boolean isRecent, Flags flags) throws MailboxException;
+    AppendResult appendMessage(InputStream msgIn, Date internalDate, Date saveDate, MailboxSession mailboxSession, boolean isRecent, Flags flags) throws MailboxException;
```

Comment:
IMO the `MessageManager` and `MessageIdManager` are responsible of setting the saveDate.

We should not leak it on this API